plugins/llvm/llvm_binary.hpp

```diff
@@ -496,11 +523,11 @@ image* create_image(const Binary* binary) {
 
 image* create(const char* data, std::size_t size) {
     StringRef data_ref(data, size);
-    MemoryBuffer* buff(MemoryBuffer::getMemBufferCopy(data_ref, "binary"));
-    OwningPtr<object::Binary> binary;
-    if (error_code ec = createBinary(buff, binary))
+    MemoryBufferRef buf(data_ref, "binary");
+    ErrorOr<std::unique_ptr<object::Binary>> binary = createBinary(buf);
+    if (std::error_code ec = binary.getError())
         llvm_binary_fail(ec);
-    return create_image(binary.get());
+    return create_image(binary->get());
```

Comment:
It looks like a use after free, that I overseen before. The pointer to the binary (both `unique_ptr` and `OwningPtr` has a shorter lifetime, than the `image`. So this code looks incorrect, if a binary is not stored, then we should pass a unique ptr, to guarantee this.